### PR TITLE
Feature multiple permissions rest

### PR DESCRIPTION
**Pull request by RetGal**
Implements #258

### CODE REVIEW

File: AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java [hunk 1]
```diff
@@ -459,6 +458,149 @@ public Integer createRestriction(String roleName, String userName, String permis
...
+                                        RestrictionEntity restriction = new RestrictionEntity();
+                                        if (createRestriction(null, userName, permissionName, resourceGroupId, null,
+                                                resourceTypePermission, contextName, action, restriction) != null) {
+                                            count++;
+                                        }
+                                    }
+                                }
+                            }
+                        } else {
+                            if (resourceTypeNames != null && !resourceTypeNames.isEmpty()) {
```
Comment (yvespp) [inline AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java]: Almost identical code of line 503+

File: AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java [hunk 2]
```diff
@@ -459,6 +458,149 @@ public Integer createRestriction(String roleName, String userName, String permis
...
+                                        count++;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                if (userNames != null && !userNames.isEmpty()) {
+                    for (String userName : userNames) {
+                        if (resourceGroupIds != null && !resourceGroupIds.isEmpty()) {
```
Comment (yvespp) [inline AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java]: Duplicate code of line 484+

File: AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java [hunk 3]
```diff
@@ -459,6 +458,149 @@ public Integer createRestriction(String roleName, String userName, String permis
...
+                                        RestrictionEntity restriction = new RestrictionEntity();
+                                        if (createRestriction(roleName, null, permissionName, null, resourceTypeName,
+                                                resourceTypePermission, contextName, action, restriction) != null) {
+                                            count++;
+                                        }
+                                    }
+                                }
+                            }
+                        } else {
+                            if (contextNames == null || contextNames.isEmpty()) {
```
Comment (yvespp) [inline AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/security/boundary/PermissionBoundary.java]: Almost identical code of line 505+

**Comment by yvespp** [comment]
@RetGal maybe something like this:
```java
public int createMultipleRestrictions(String roleName, List<String> userNames, List<String> permissionNames, List<Integer> resourceGroupIds, List<String> resourceTypeNames, 
										ResourceTypePermission resourceTypePermission, List<String> contextNames, List<Action> actions) throws AMWException {
	// ensure that we loop at least once over every element
	if (userNames == null || userNames.size() == 0) {
		userName = new ArrayList()
		userName.add(null)
	}
	if (permissionNames == null || permissionNames.size() == 0) {
		permissionNames = new ArrayList()
		permissionNames.add(null)
	}
	....

	for (String permissionName : permissionNames) {
		for (String userName : userNames) {
			for (String resourceGroupId : resourceGroupIds) {
			....
				RestrictionEntity restriction = new RestrictionEntity();
				if (createRestriction(roleName, userName, permissionName, resourceGroupId, resourceTypeName,
					resourceTypePermission, contextName, action, restriction) != null) {
					count++;
				}
```

With this no if/else is needed in the loop.

**Comment by RetGal** [comment]
@yvespp Refactored as suggested

**Pull request merged**